jetstream/tests/test_statistics.py

```diff
@@ -102,6 +108,48 @@ def test_binomial(self):
         assert difference.point - 0.2 < 1e-5
         assert difference.lower and difference.upper
 
+    def test_binomial_pairwise_branch_comparisons(self, experiments):
+        stat = Binomial()
+        test_data = pd.DataFrame(
+            {
+                "branch": ["treatment"] * 10 + ["control"] * 10 + ["foo"] * 10,
+                "value": [False] * 7
+                + [True] * 3
+                + [False] * 5
+                + [True] * 5
+                + [False] * 5
+                + [True] * 5,
+            }
+        )
+        results = stat.apply(
+            test_data, "value", experiments[1], AnalysisBasis.ENROLLMENTS, "all"
+        ).__root__
+
+        branch_results = [r for r in results if r.comparison is None]
+        treatment_result = [r for r in branch_results if r.branch == "treatment"][0]
+        control_result = [r for r in branch_results if r.branch == "control"][0]
+        assert treatment_result.point < control_result.point
+        assert treatment_result.point - 0.7 < 1e-5
+
+        difference = [r for r in results if r.comparison == "difference"][0]
+        assert difference.point - 0.2 < 1e-5
+        assert difference.lower and difference.upper
+
+        comparison_branches = [
+            (r.comparison_to_branch, r.branch, r.comparison) for r in results
+        ]
+        assert (None, "control", None) in comparison_branches
+        assert (None, "foo", None) in comparison_branches
+        assert (None, "treatment", None) in comparison_branches
+        assert ("treatment", "control", "difference") in comparison_branches
+        assert ("treatment", "control", "relative_uplift") in comparison_branches
+        assert ("control", "treatment", "difference") in comparison_branches
+        assert ("control", "treatment", "relative_uplift") in comparison_branches
+        assert ("foo", "control", "difference") in comparison_branches
+        assert ("foo", "control", "relative_uplift") in comparison_branches
+        assert ("control", "foo", "difference") in comparison_branches
+        assert ("control", "foo", "relative_uplift") in comparison_branches
+
```

Comment:
Can we add lines here to test that pairs of `treatment` and `foo` are in the results? we have `(control, treatment)` and `(control, foo)` but are missing `treatment,foo`

Reply:
Whoops, good catch! Added.

Reply:
Ok I decided to just refactor the test a little bit to test that the two lists are equal instead of individually testing every element in the list.

Reply:
Awesome